docs/rules/no-restricted-amd-modules.md

```diff
@@ -0,0 +1,101 @@
+# Disallow specific amd modules (no-restricted-amd-modules)
+
+Disallowing usage of specific AMD modules can be useful if you want to limit the available methods a developer can use. For example, you can block usage of the `foo` module if you want to force using a more performant module named `to-foo-to-furious`.
+
+## Rule Details
+
+This rule allows you to specify modules that you don’t want to use in your application.
+
+## Options
+
+The rule takes one or more strings as options: the names of restricted modules.
+
+```json
+"no-restricted-amd-modules": ["error", "foo-module", "bar-module"]
+```
+
+It can also take an object with lists of `paths` and gitignore-style `patterns` strings.
+
+```json
+"no-restricted-amd-modules": ["error", { "paths": ["foo-module", "bar-module"] }]
+```
+
+```json
+"no-restricted-amd-modules": ["error", {
+    "paths": ["foo-module", "bar-module"],
+    "patterns": ["foo-module/private/*", "bar-module/*","!baz-module/good"]
+}]
+```
+
+You may also specify a custom message for any paths you want to restrict as follows:
+
+```json
+"no-restricted-amd-modules": ["error", [{
+  "name": "foo-module",
+  "message": "Please use bar-module instead."
+  }]
+]
+```
+
+or like this:
+
+```json
+"no-restricted-amd-modules": ["error",{
+"paths":[{
+  "name": "foo-module",
+  "message": "Please use bar-module instead."
+  }]
+}]
+```
+
+The custom message will be appended to the default error message. Please note that you may not specify custom error messages for restricted patterns as a particular module may match more than one pattern.
+
+
+## Examples
+
+Examples of **incorrect** code for this rule  with sample `"foo", "bar", "baz"` restricted modules:
```

Comment:
This line doesn't match the configuration comments used in the actual examples. Is that intentional?

Reply:
Fixed